### DISCUSSION TITLE
how to get the current logged in user id

### DISCUSSION THREAD

**Discussion by heleopless50** [serenity-is/Serenity#6758] [Help] [answered]
[ServiceLookupEditor(service: "Default/WarehouseUsers/List",idField :"WarehouseId",textField:"WarehouseId",FilterField ="UserId",FilterValue = /*       i want to get the current user id here          */]
        public Int64? WarehouseId
        {
            get { return Fields.WarehouseId[this]; }
            set { Fields.WarehouseId[this] = value; }
        }

i have this property in Purchase row and i get the ware houses for current logged in user 
how to pass the current logged in user id in the filter value 
thank you

**Comment by reach2rv** [comment]
You need to handle this on your list handler by overriding Applyfilters

**Comment by heleopless50** [comment]
i dont understand 
how can i do this ?

**Reply by heleopless50** [reply, marked as answer]
i solved it by delegating the getting of warehouses specific to a user to an end point then i call it using service lookup editor 

        public ListResponse<MyRow> ListUser(IDbConnection connection, ListRequest request,
            [FromServices] IWarehouseUsersListHandler handler)
        {
            var userName = User.Identity.Name;
            var currentUser = userService.ByUsername(userName);        
            var userId = Convert.ToInt32(currentUser.Id??"0");
            var listHandler = handler.List(connection, request);
            listHandler.Entities  = handler.List(connection, request).Entities.FindAll(item => item.UserId == userId);
    
            return listHandler;


        }

**Reply by heleopless50** [reply]
[ServiceLookupEditor(service: "Default/WarehouseUsers/ListUser",idField :"WarehouseId",textField:"WarehouseId",IncludeColumns = new String[]{"UserId"})]
        public Int64? WarehouseId
        {
            get { return Fields.WarehouseId[this]; }
            set { Fields.WarehouseId[this] = value; }
        }
